File: 2545-sort-the-students-by-their-kth-score/2545-sort-the-students-by-their-kth-score.java
class Solution {
    public int[][] sortTheStudents(int[][] score, int k) {
        int n = score.length;
        PriorityQueue<int[]> preq = new PriorityQueue<>(
            (a, b) -> Integer.compare(b[0], a[0])
        );
        
        for (int i = 0; i < n; i++) {
            preq.add(new int[]{score[i][k], i});
        }
        int[][] result = new int[n][];
        int idx = 0;
        
        while (!preq.isEmpty()) {
            int[] x = preq.poll();
            result[idx++] = score[x[1]];
        }
        return result;
    }
}

File: 2545-sort-the-students-by-their-kth-score/2545-sort-the-students-by-their-kth-score.cpp
class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>>& score, int k) {
        int n = score.size();

        priority_queue<pair<int,int>> preq;

        for(int i =0 ;i<n ; i++) {
            preq.push({score[i][k],i});
        }

        vector<vector<int>> result;

        while(!preq.empty()) {
            auto x = preq.top();
            preq.pop();

            result.push_back(score[x.second]);
        }
        return result;
    }
};